docs/contributor/image_mode.md

```diff
@@ -536,3 +536,44 @@ It is now possible to access images using the manifest name with the `latest` ta
 (e.g. `quay.io/myorg/mypath/microshift-4.17-bootc:latest`). The image for the
 current platform will automatically be pulled from the registry if it is part of
 the manifest list.
+
+## Appendix C: The rpm-ostree to Image Mode Upgrade Procedure
+
+Refer to RHEL documentation for generic instructions on upgrading `rpm-ostree`
+systems to Image Mode. The upgrade process should be planned carefully considering
+the following guidelines:
+* Follow instructions in RHEL documentation for converting `rpm-ostree` blueprints to
+  Image Mode container files
+* Consider using [rpm-ostree compose container-encapsulate](https://coreos.github.io/rpm-ostree/container/#converting-ostree-commits-to-new-base-images)
+  to experiment with Image Mode based on the existing `ostree` commits
+* Invest in defining a proper container build pipeline for fully adopting Image Mode
+
+If reinstalling MicroShift devices from scratch is not an option, read the remainder
+of this section that outlines the upgrade details specific to MicroShift.
+
+Upgrading existing systems during the transition from `rpm-ostree` to Image Mode
+may pose the challenge of [UID / GID Drift](https://github.com/bootc-dev/bootc/issues/673)
+because the existing `rpm-ostree` and the new Image Mode images are not derived
+from the same parent image.
+
+One way of working around this issue is to add `systemd` units that run before the
+affected system services and apply the necessary fixes.
```

Comment:
Have you thought about "how" long-term is this workaround? Do you think these need to be carried for eternity or are they needed only for the first upgrade rpm-ostree -> bootc?

Reply:
That's a good point. Let me add a note about this.